website/docs/api/kb.md

```diff
@@ -4,169 +4,88 @@ teaser:
-| --------------- | --------------------------------------------------------------------------------- |
-| `alias`         | The textual mention or alias. Can not be the empty string. ~~str~~                |
-| `entities`      | The potential entities that the alias may refer to. ~~Iterable[Union[str, int]]~~ |
-| `probabilities` | The prior probabilities of each entity. ~~Iterable[float]~~                       |
-
-## KnowledgeBase.\_\_len\_\_ {#len tag="method"}
-
-Get the total number of entities in the knowledge base.
-
-> #### Example
->
-> ```python
-> total_entities = len(kb)
-> ```
-
-| Name        | Description                                           |
-| ----------- | ----------------------------------------------------- |
-| **RETURNS** | The number of entities in the knowledge base. ~~int~~ |
-
-## KnowledgeBase.get_entity_strings {#get_entity_strings tag="method"}
-
-Get a list of all entity IDs in the knowledge base.
-
-> #### Example
->
-> ```python
-> all_entities = kb.get_entity_strings()
-> ```
-
-| Name        | Description                                               |
-| ----------- | --------------------------------------------------------- |
-| **RETURNS** | The list of entities in the knowledge base. ~~List[str]~~ |
-
-## KnowledgeBase.get_size_aliases {#get_size_aliases tag="method"}
-
-Get the total number of aliases in the knowledge base.
+Given a certain textual mention as input, retrieve a list of candidate entities
+of type [`Candidate`](/api/candidate).
 
 > #### Example
 >
 > ```python
-> total_aliases = kb.get_size_aliases()
+> from spacy.lang.en import English
+> nlp = English()
+> doc = nlp("Douglas Adams wrote 'The Hitchhiker's Guide to the Galaxy'.")
+> candidates = kb.get_candidates(doc[0:2])
 > ```
 
-| Name        | Description                                          |
-| ----------- | ---------------------------------------------------- |
-| **RETURNS** | The number of aliases in the knowledge base. ~~int~~ |
-
-## KnowledgeBase.get_alias_strings {#get_alias_strings tag="method"}
-
-Get a list of all aliases in the knowledge base.
-
-> #### Example
->
-> ```python
-> all_aliases = kb.get_alias_strings()
-> ```
+| Name        | Description                                                          |
+| ----------- | -------------------------------------------------------------------- |
+| `mention`   | The textual mention or alias. ~~Span~~                               |
+| **RETURNS** | An iterable of relevant `Candidate` objects. ~~Iterable[Candidate]~~ |
 
-| Name        | Description                                              |
-| ----------- | -------------------------------------------------------- |
-| **RETURNS** | The list of aliases in the knowledge base. ~~List[str]~~ |
+## KnowledgeBase.get_candidates_batch {#get_candidates_batch tag="method"}
 
-## KnowledgeBase.get_alias_candidates {#get_alias_candidates tag="method"}
```

Comment:
`kb#get_alias_candidates` is referred to from the `v3.md` file, so we probably want to keep the section and refer to the update of functionality & explain from which version it happened.

Reply:
Added a note, let me know what you think. One concern: it might look like this method is still available, since it's in the docs. I added an infobox, but still. I'm ok with leaving it in until v4 though.